pep_sphinx_extensions/__init__.py

```diff
@@ -37,12 +38,37 @@ def _depart_maths():
     pass  # No-op callable for the type checker
 
 
-def _update_config_for_builder(app: Sphinx):
+def _check_external_builder(app: Sphinx, conf: Config) -> None:
+    if "internal_builder" not in app.tags:  # internal_builder exists if Sphinx is run by build.py
+        conf.html_context = {"external_builder": True}
+    else:
+        conf.html_context = {"external_builder": False}
+    app.tags.remove("internal_builder")
+
+    if conf.html_context["external_builder"]:
+        app.connect("build-finished", _post_build)  # Post-build tasks
+
+
+def _update_config_for_builder(app: Sphinx) -> None:
+    if app.config.html_context["external_builder"]:
+        app.builder.copysource = False  # Prevent unneeded source copying - we link direct to GitHub
+        app.builder.search = False  # Disable search
+
     if app.builder.name == "dirhtml":
         config.pep_url = f"../{config.pep_stem}"
```

Comment:
There's no config variable at this point?

Reply:
As in `config.pep_url`? That (somewhat hackily) updates the `pep_url` variable in `pep_sphinx_extensions.config` -- as I found no better way of dynamically specifying the URLs for file vs directory based builds, given both are wanted.

Reply:
Ah, `from pep_sphinx_extensions import config` -- well, that's confusing. ;)

Reply:
Sorry! I don't like double redundant names (`pep_config` etc), although I guess the module could be called `constants` or something to make it clearer